test/common_nn.py

```diff
@@ -3136,18 +3138,42 @@ def padding3d_circular(input, pad):
         check_sum_reduction=True,
         desc='dim_is_3',
     ),
+    dict(
+        module_name='PoissonNLLLoss', # Default is log_input=True, full=False
+        input_size=(2, 3, 4, 5),
+        target_fn=lambda: torch.randn(2, 3, 4, 5).floor_().abs_(),
+        reference_fn=lambda i, t, _: (i.exp() - t.mul(i)).mean(),
+        desc='no_full_loss',
+        check_forward_only=False,
```

Comment:
I found that `criterion_tests` doesn't test backward by default. So, I added `check_forward_only=False` explicitly here for easier code review.
But, I suppose, such default is intended for faster test execution time, so I'll remove this if it's desired so.

Reply:
@mruberry do you know what is our take on this? Should we set this back to the default before merging?

Reply:
It looks like PyTorch tests backwards in module_tests and not criterion_tests (per @hi-ogawa's comment). Without a compelling reason I wouldn't change this behavior. NLLLoss is also tested by the module tests. (Unfortunately our test suite is a confusing mess of redundancy.)

Reply:
I guess the point here is that since the function itself is autodiffed, there is no point to check the gradient computations.
@hi-ogawa could you set this flag back to the default value (remove the line).

Reply:
@mruberry @albanD Thanks a lot for the review. Right, `poissonnllloss_no_reduce_test` (Line 280) is included in `new_module_tests` instead of `new_criterion_tests` and that one actually checks backward. Okay, I removed `check_forward_only=False` to align with other tests here and I hope it's ready to merge now.